### PR TITLE
Update developer documentation with new 8.0 images

### PR DESCRIPTION
**Pull request by sns-seb**
- [X] explain directory structure by edition
- [X] explain new (and secured) way to build images from non public artifacts
- [X] explain branch "branch-8"
- [X] add details about ITs
- [X] update release guidelines

### CODE REVIEW

**Comment by sns-seb** [comment]
@pierre-guillot-sonarsource since you created the doc PR, do you see any other relevant information/recommendation to add to the part about it?

File: develop.md [hunk 1]
```diff
@@ -1,49 +1,195 @@
...
-    4. Verify the Pull Request passes the automated tests (visible in the status of the PR)
+New non-major version
+---------------------
 
-3. Wait for the Pull Requests to get merged, and verify the updated page on [Docker Hub][hub]
+As of today, we publish only a single version of SQ for a given major version in the "master" branch.
 
-FAQ
----
+New new image is therefor created for non-major versions of SQ. Instead, existing images are updated.
```
Comment (pierre-guillot-sonarsource) [inline develop.md]:
```suggestion
New image is therefor created for non-major versions of SQ. Instead, existing images are updated.
```

Comment (michal-duda-sonarsource) [inline develop.md]:
```suggestion
New new image is therefore created for non-major versions of SQ. Instead, existing images are updated.
```

Comment (michal-duda-sonarsource) [reply develop.md]: `Instead`?

File: develop.md [hunk 2]
```diff
@@ -1,49 +1,195 @@
...
+=========================================
+
+Dogfooding
+----------
+
+SonarQube Docker images are dogfooded.
+
+This implies that Docker images are produced for every artifact produced from the dogfood branch of SonarQube (see the [dogfood_docker_build_task](https://github.com/SonarSource/sonar-enterprise/blob/master/.cirrus.yml#L263)).
+
+Docker images of SonarQube are built from publicly available artifacts, which means the DockerFile can simply download them from the public place they are hosted (`binaries.sonarsource.com`).
```
Comment (michal-duda-sonarsource) [inline develop.md]: It's always named just`Dockerfile`, no need to uppercase the `file` part (needs changing in a few places).

File: develop.md [hunk 3]
```diff
@@ -1,49 +1,195 @@
...
------------
+Dogfooding and images for local artifacts
+=========================================
+
+Dogfooding
+----------
+
+SonarQube Docker images are dogfooded.
+
+This implies that Docker images are produced for every artifact produced from the dogfood branch of SonarQube (see the [dogfood_docker_build_task](https://github.com/SonarSource/sonar-enterprise/blob/master/.cirrus.yml#L263)).
```
Comment (michal-duda-sonarsource) [inline develop.md]: Link is quite useful now but will become outdated very quickly. Not a big deal and I have no strong opinion but maybe just name of the job + the link to cirrus `conf` would be better?

File: develop.md [hunk 4]
```diff
@@ -1,49 +1,195 @@
...
+4. an IT ensuring that the 6 DockerFiles are up to date with the "template" and the script (to prevent dev from forgetting to commit up-to-date DockerFiles)
+
+With this idea:
 
-- https://github.com/docker-library/docs
-- https://github.com/docker-library/official-images
+1. trust in dogfooding Docker images representatives of the one which will end up as the official images moves from the developer doing the right changes when cleaning the "version" branch to the script and the template
+  * no more human based last minute changes is an obvious improvement
+  * it's all commited so an error is easy to track
+2. there is no more need for a "version" branch
```
Comment (michal-duda-sonarsource) [inline develop.md]: `... no longer a need ...`

File: develop.md (hunk 1 above)
Comment (sns-seb) [reply develop.md]: "New new" was intended to be "No new"

Indeed, with that typo, both sentences hardly make any sense

Comment (sns-seb) [reply develop.md]: "New new" should actually be "No new"

fixing

File: develop.md (hunk 3 above)
Comment (sns-seb) [reply develop.md]: I thought about it

I considered that the link is correct now and this is convenient.
If line becomes incorrect in the future, the link will still point to the correct file. A bit less convenient, the result will be the same as providing the link without the line but at least it's been more convenient for a while

Comment (sns-seb) [reply develop.md]: I thought about it

I considered that the link is correct now and this is convenient.
If line becomes incorrect in the future, the link will still point to the correct file. A bit less convenient, the result will be the same as providing the link without the line but at least it's been more convenient for a while

**Pull request merged**